apps/src/StudioApp.js

```diff
@@ -1813,6 +1816,23 @@ function runButtonClickWrapper(callback) {
     $(window).trigger('run_button_pressed');
     $(window).trigger('appModeChanged');
   }
+
+  // inform Blockly that the run button has been pressed
+  if (window.Blockly && Blockly.mainBlockSpace) {
+    var customEvent;
+    // createEvent functionality is officially deprecated in favor of
+    // the event constructor, but some older browsers do not yet support
+    // event constructors. Attempt to use the new functionality, fall
+    // back to the old if it fails.
+    try {
+      customEvent = new Event(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED);
+    } catch (e) {
+      customEvent = document.createEvent('Event');
+      customEvent.initEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED, true, false);
+    }
+    Blockly.mainBlockSpace.getCanvas().dispatchEvent(customEvent);
```

Comment:
Also perfectly reasonable to skip if you've done barrage of x-browser tests on this and would prefer to hold off on doing another just to move a bit of logic around :)